packages/xo-web/src/xo-app/backup-ng/new/index.js

```diff
@@ -186,20 +186,40 @@ export default decorate([
           }
         }
 
-        await createBackupNgJob({
-          name: state.name,
-          mode: state.isDelta ? 'delta' : 'full',
-          compression: state.compression,
-          schedules: mapValues(
+        let schedules, settings
+        if (!isEmpty(state.schedules)) {
+          schedules = mapValues(
             state.schedules,
             ({ id, ...schedule }) => schedule
-          ),
-          settings: normalizeSettings({
+          )
+          settings = normalizeSettings({
             settings: state.settings,
             exportMode: state.exportMode,
             copyMode: state.copyMode,
             snapshotMode: state.snapshotMode,
-          }).toObject(),
+          }).toObject()
+        } else {
+          const id = generateId()
+          schedules = {
+            [id]: DEFAULT_SCHEDULE,
+          }
+          settings = {
+            [id]: {
+              copyRetention: state.copyMode ? DEFAULT_RETENTION : undefined,
+              exportRetention: state.exportMode ? DEFAULT_RETENTION : undefined,
+              snapshotRetention: state.snapshotMode
+                ? DEFAULT_RETENTION
+                : undefined,
+            },
+          }
+        }
+
+        await createBackupNgJob({
+          name: state.name,
+          mode: state.isDelta ? 'delta' : 'full',
+          compression: state.compression,
+          schedules,
+          settings,
           remotes:
             state.deltaMode || state.backupMode
               ? constructPattern(state.remotes)
```

Comment:
try to destructure the state in the begining of `createJob` effect.
`const { isDelta, compression, deltaMode, backupMode, copyMode, exportMode, snapshotMode } = state`

Reply:
I don't see the advantage of destructing properties here. It will just add some lines.